akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -179,22 +180,38 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
               incomingStreams(streamId).inlet.foreach(_.cancel())
 
             case WindowUpdateFrame(streamId, increment) ⇒
-              incomingStreams(streamId).outboundWindowLeft += increment
-              debug(f"outbound window for [$streamId%3d] is now ${incomingStreams(streamId).outboundWindowLeft}%10d after increment $increment%6d")
-              bufferedFrameOut.tryFlush()
+              try {
+                incomingStreams(streamId).outboundWindowLeft += increment
+
+                println(s"incomingStreams(streamId) = ${incomingStreams(streamId)} ( added + $increment ) ")
+                debug(f"outbound window for [$streamId%3d] is now ${incomingStreams(streamId).outboundWindowLeft.value}%10d after increment $increment%6d")
+                bufferedFrameOut.tryFlush()
+              } catch {
+                case ex: ArithmeticException ⇒
```

Comment:
Not sure if I like the exception handling here. Let's try to centralize window handling first (multiplexer PR goes already in that direction) before checking it.